controls/V-72049.rb

```diff
@@ -45,19 +51,51 @@
 environment variables."
   tag "fix_id": "F-78401r1_fix"
 
-  # @todo - test for values more restrictive than 077
-  file_lines = command('grep -i -s umask /home/*/.*').stdout.split("\n")
-  if !file_lines.nil? and !file_lines.empty?
-    file_lines.each do |curr_line|
-      file_name = curr_line.split(':').first
-      describe command("grep -i umask #{file_name}") do
-        its('stdout.strip') { should match %r{^umask\s+.*077} }
-      end
+  # Get all interactive users
+  ignore_shells = non_interactive_shells.join('|')
+  
+  # Get home directory for users with UID >= 1000 or UID == 0 and support interactive logins.                                                              
+  findings = Set[]
+  dotfiles = Set[]
+  umasks = {}
+  interactive_users = users.where{ !shell.match(ignore_shells) && (uid >= 1000 || uid == 0)}.entries
+
+  # For each user, build and execute a find command that identifies initialization files                                                                   
+  # in a user's home directory.                                                                                                                            
+  interactive_users.each do |u|
+    # Get user's initialization files
+    dotfiles = dotfiles + command("find #{u.home} -xdev -maxdepth 2 ( -name '.*' ! -name '.bash_history' ) -type f").stdout.split("\n")
+    
+    # Get user's umask
+    umasks.store(u.username,command("su -c 'umask' -l #{u.username}").stdout.chomp("\n"))
+  end
+
+  # Check all local initialization files to see whether or not they are less restrictive than 077.
+  dotfiles.each do |df|
+    if file(df).more_permissive_than?("0077")
+      findings = findings + df
     end
-  else
-    describe "No interactive files with a less restrictive umask were found." do
-      subject { file_lines.nil? or file_lines.empty? }
-      it { should eq true }
+  end
+  
+  # Report on any interactive files that are less restrictive than 077.
```

Comment:
I thought that `027` was allowed?

Reply:
027 is more restrictive than 077 so it is allowed.

Reply:
Heh..internal bitmask fail